google/cloud/bigquery/_helpers.py

```diff
@@ -114,6 +126,38 @@ def _int_from_json(value, field):
         return int(value)
 
 
+def _interval_from_json(
+    value: Optional[str], field
+) -> Optional[relativedelta.relativedelta]:
+    """Coerce 'value' to an interval, if set or not nullable."""
+    if not _not_null(value, field):
+        return
+    if value is None:
+        raise TypeError(f"got {value} for REQUIRED field: {repr(field)}")
+
+    parsed = _INTERVAL_PATTERN.match(value)
```

Comment:
Check for parsed being `None`.  Something like "Invalid format for interval data, ..." is better that `None has no attribute group`.